QualifiedTeachersApi/src/QualifiedTeachersApi/DataStore/Crm/DataverseAdapter.SetIttResultForTeacher.cs

```diff
@@ -325,28 +325,57 @@ public async Task<SetIttResultForTeacherLookupResult> LookupData()
             Guid ittProviderId)
         {
             // Find an ITT record for the specified ITT Provider.
-            // The record should be at the InTraining,WithDrawn or Deferred status unless the programme is 'assessment only',
-            // in which case the status should be UnderAssessment, WithDrawn or Deferred.
-
-            var inTrainingForProvider = ittRecords
-                .Where(r => (r.dfeta_ProgrammeType != dfeta_ITTProgrammeType.AssessmentOnlyRoute && (r.dfeta_Result == dfeta_ITTResult.InTraining || r.dfeta_Result == dfeta_ITTResult.Withdrawn || r.dfeta_Result == dfeta_ITTResult.Deferred) ||
-                            (r.dfeta_ProgrammeType == dfeta_ITTProgrammeType.AssessmentOnlyRoute && (r.dfeta_Result == dfeta_ITTResult.UnderAssessment || r.dfeta_Result == dfeta_ITTResult.Withdrawn || r.dfeta_Result == dfeta_ITTResult.Deferred))))
+            // if ProgrammeType is AssessmentOnlyRoute, result should be UnderAssessment,Withdrawn or Deferrred otherwise
+            // record should be InTraining,WithDrawn or Deferred
+            List<dfeta_initialteachertraining> matching = new List<dfeta_initialteachertraining>();
 
+            var activeForProvider = ittRecords
```

Comment:
Nit: we have a single `Where()` in the `UpdateTeacher` method and would be good to keep them aligned.